### DISCUSSION TITLE
How to use timeouts in the configuration?

### DISCUSSION THREAD

**Discussion by jvalrog** [mikro-orm/mikro-orm#3926] [Q&A] [answered]
**Describe the bug**
So far I'm doing this, but doesn't work since the initial connection still timeout after 10 seconds (mysql2 default timeout):

```
MikroOrmModule.forRootAsync({
      contextName: BandaanchaGGCCToken,
      inject: [ConfigService],
      useFactory: (config: ConfigService<AppConfig>) => ({
        host: config.get('DB_RADIUS_HOST'),
        port: +config.get('DB_RADIUS_PORT'),
        user: config.get('DB_RADIUS_USER'),
        password: config.get('DB_RADIUS_PASSWORD'),
        dbName: config.get('DB_RADIUS_NAME'),
        type: config.get('DB_RADIUS_ADAPTER'),
        debug: config.get('NODE_ENV') !== 'production',
        forceUtcTimezone: true,
        driverOptions: {
          connectTimeout: 60000,
          acquireTimeout: 60000,
        },
        autoLoadEntities: true,
        registerRequestContext: false,
        findOneOrFailHandler: (entityName: string) =>
          new NotFoundException(`${entityName} not found`),
      }),
    }),
```

**Versions**

| Dependency | Version |
| node | 16.18.1 |
| typescript | 4.7.4 |
| mikro-orm | 5.4.0 |
| mikro-orm/mysql | 5.4.0 |

**Comment by B4nan** [comment, marked as answer]
Where did you find those options? `driverOptions` is used to pass options to knex, are those options of knex? You probably want to use [acquireConnectionTimeout](https://knexjs.org/guide/#acquireconnectiontimeout) instead.

**Reply by jvalrog** [reply]
oh I see, will try that, thanks. But it says 60 seconds as default so that may be not my real problem since the connection timeout at 10 seconds.

There is an "insecureAuth" option in knex that I'd like to try with Mikro-Orm since our db is a bit old and that might help.

**Reply by jvalrog** [reply]
solved the issue with this config:

```
driverOptions: {
          connection: {
            insecureAuth: true,
            connectTimeout: 60000,
          },
        },
```
